src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -50,6 +50,32 @@
     </PropertyGroup>
   </Target>
 
+  <!--
+      Target to disable analyzers via an MSBuild property 'RunAnalyzers'. 
+      Note that we remove analyzers only for non-design time builds.
+      Design time builds are background builds inside Visual Studio,
+      see details here: https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md
+      We do not remove analyzers for design time builds to allow users to view the installed analyzers
+      in 'Analyzers' node in the solution explorer and also execute analyzers on-demand with "Run Code Analysis" command.
+      Instead, for design time builds, the VS IDE will read this MSBuild property and skip analyzer execution if it is 'false'.
+  -->
+  <Target Name="_DisableAnalyzers"
+          BeforeTargets="CoreCompile"
+          Condition="'$(RunAnalyzers)' != 'true'">
+    <PropertyGroup>
+      <!-- Determine if this is a design time build: https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md#determining-whether-a-target-is-running-in-a-design-time-build -->
+      <_IsAnyDesignTimeBuild Condition="'$(DesignTimeBuild)' == 'true' or '$(BuildingProject)' != 'true'">true</_IsAnyDesignTimeBuild>
+
+      <!-- Fallback 'RunAnalyzers' to 'RunAnalyzersDuringBuild', the latter property allows users to disable analyzers only for non-design time builds. -->
+      <RunAnalyzers Condition="'$(RunAnalyzers)' == '' and '$(_IsAnyDesignTimeBuild)' != 'true'">$(RunAnalyzersDuringBuild)</RunAnalyzers>
```

Comment:
The impact of this statement is that it will change the property value for any targets that run after this. Is that the intent? If so wouldn't it be cleaner to do this in a property group that executes before targets are evaluated? 

Rather than changing the global value though should we create a new local property, say `<_RunAnalyzers>` and use that in the target? Basically change local state not global.

Reply:
Using a new property `_RunAnalyzers` sounds good. I will switch.